Frameworks/CoreText/CTRun.mm

```diff
@@ -24,52 +24,36 @@
 #import "CoreTextInternal.h"
 #import "CGContextInternal.h"
 #import <CppUtils.h>
-
-#include <numeric>
-#include <algorithm>
-#include <functional>
+#import <numeric>
+#import <algorithm>
+#import <functional>
 
 const CFStringRef kCTBackgroundStrokeColorAttributeName = CFSTR("kCTBackgroundStrokeColorAttributeName");
 const CFStringRef kCTBackgroundFillColorAttributeName = CFSTR("kCTBackgroundFillColorAttributeName");
 const CFStringRef kCTBackgroundCornerRadiusAttributeName = CFSTR("kCTBackgroundCornerRadiusAttributeName");
 const CFStringRef kCTBackgroundLineWidthAttributeName = CFSTR("kCTBackgroundLineWidthAttributeName");
 
-@implementation _CTRun : NSObject
-- (void)dealloc {
-    _stringFragment = nil;
-    _attributes = nil;
-    _SafeRelease(&_dwriteGlyphRun.fontFace);
-    delete[] _dwriteGlyphRun.glyphIndices;
-    delete[] _dwriteGlyphRun.glyphAdvances;
-    delete[] _dwriteGlyphRun.glyphOffsets;
-    [super dealloc];
-}
-
-- (id)init {
-    if (self = [super init]) {
-        _attributes.attach([NSMutableDictionary new]);
-        _textMatrix = CGAffineTransformIdentity;
-    }
-
-    return self;
+// We need to include DWriteWrapper_CoreText.h for _CloneDWriteGlyphRun,
+// which is located in the Frameworks\CoreText folder
+// Given how the earlier implementation is accessing and assigning internal variables in different parts of this package
+// struct __CTRun has to be made public via CoreTextInternal.h (Frameworks\include). Therefore __CTRun(const CTRunRef& other)'s
+// implementation needs to be moved here.
+__CTRun::__CTRun(const CTRunRef& other)
+    : _attributes(CFDictionaryCreateMutableCopy(kCFAllocatorSystemDefault, CFDictionaryGetCount(other->_attributes), other->_attributes)),
+      _range(other->_range),
+      _stringIndices(other->_stringIndices),
+      _glyphOrigins(other->_glyphOrigins),
+      _glyphAdvances(other->_glyphAdvances),
+      _relativeXOffset(other->_relativeXOffset),
+      _relativeYOffset(other->_relativeYOffset),
+      _stringFragment(CFStringCreateCopy(kCFAllocatorDefault, other->_stringFragment)) {
+    _CloneDWriteGlyphRun(&(other->_dwriteGlyphRun), &_dwriteGlyphRun);
 }
 
-- (instancetype)copyWithZone:(NSZone*)zone {
-    _CTRun* ret = [_CTRun new];
-    ret->_attributes.attach([_attributes copy]);
-    ret->_range = _range;
-    ret->_stringFragment.attach([_stringFragment copy]);
-    _CloneDWriteGlyphRun(&_dwriteGlyphRun, &(ret->_dwriteGlyphRun));
-    ret->_stringIndices = _stringIndices;
-    ret->_glyphOrigins = _glyphOrigins;
-    ret->_glyphAdvances = _glyphAdvances;
-    ret->_relativeXOffset = _relativeXOffset;
-    ret->_relativeYOffset = _relativeYOffset;
-    return ret;
+CTRunRef _CTRunCreate() {
```

Comment:
abstractions make life easier for maintainers.
Just because it's already bad, i don't think it should go on :p
I do realize that i introduced one and that will get fixed up. 

---
In reply to: [111501891](https://github.com/Microsoft/WinObjC/pull/2511#discussion_r111501891) [](ancestors = 111501891)